### PR TITLE
waf: ensure ENABLE_SCRIPTING is always defined

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/ardupilotwaf/boards.py [hunk 1]
```diff
@@ -48,6 +48,10 @@ def srcpath(path):
         env.SRCROOT = srcpath('')
         self.configure_env(cfg, env)
 
+        env.DEFINES.update(
+            ENABLE_SCRIPTING = 0,
```
Comment (tridge) [inline Tools/ardupilotwaf/boards.py:52]: this would break lots of code, as it uses "#ifdef" now

**Review by tridge — CHANGES_REQUESTED** [review]

File: Tools/ardupilotwaf/boards.py (hunk 1 above)
Comment (peterbarker) [reply Tools/ardupilotwaf/boards.py:52]: Patches have been added for that.

**Comment by peterbarker** [comment]
@tridge should we take this opportunity to move to `AP_SCRIPTING_ENABLED`?

**Comment by tridge** [comment]
> @tridge should we take this opportunity to move to `AP_SCRIPTING_ENABLED`?

yes, I think so

**Comment by peterbarker** [comment]
Renamed `ENABLE_SCRIPTING` to `AP_SCRIPTING_ENABLED`

**Review by tridge — APPROVED** [review]

**Comment by magicrub** [comment]
This is a good change that I support. That said, it conflicts with another branch I was planning on making a PR for that removes half of them :)

**Pull request merged**